qcp/gates.py

```diff
@@ -194,14 +197,12 @@ def _generic_control(size: int, controls: List[int],
     # EG: controls = [0, 2, 4]
     # corresponds to 0, 4, 16 as numbers,
     # bitmask is 10101 in binary notation
-    mask = sum(2**c for c in set(controls))
-    invertor = sum(2**i for i in range(size))
-    flip_mask = mask ^ invertor
+    mask = sum(2 ** c for c in set(controls))
 
-    target_bit = 2**target
+    target_bit = 2 ** target
 
     for i in range(0, n):
-        condition = (i & target_bit) == target_bit and i ^ mask == flip_mask
+        condition = (i & target_bit) == target_bit and i & mask >= mask
```

Comment:
For the equivalent on control_x(), the check for `>= mask` is performed in the `if condition` check. I don't know if we want to mirror that style here?

Reply:
Do you mean something like:
`condition1 = (i & target_bit) 
 condition2 = i & mask
if condition1 == target_bit and condition2 >= mask:
`

Reply:
that did not indent the way i thought it would